test/index.js

```diff
@@ -16,6 +16,22 @@ describe('lib: run-on-worker', function () {
     await expect(request).to.eventually.be.rejected.to.include({ name: 'WorkerError', message: 'bang!!!', code: 42 });
   });
 
+  it('allow for progress updates', function (done) {
```

Comment:
wondering if it might not be better to add progress events to the default worker helper and test with sinon instead?
```js
// in test/_worker.js
default:
  process.send(JSON.stringify({ progress: 42 }));
  process.send(JSON.stringify({ progress: 43 }));
  process.send(JSON.stringify({ response: message }));
  break;

// and then the test becomes...
it('should emit progress updates', async function () {
  const message = { progress: 'test' };
  const progress = sinon.stub();
  const response = await runOnWorker(workerFile, message, progress);
  expect(progress).to.have.been.calledTwice();
  expect(progress.firstCall).to.have.been.calledWith(42);
  expect(progress.secondCall).to.have.been.calledWith(43);
  expect(response).to.deep.equal(message);
});
```
this way you are testing the whole process with progress events and not just the progress events being returned.